sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -216,7 +220,35 @@ public void finish(@Nullable SpanStatus status) {
       if (transactionFinishedCallback != null) {
         transactionFinishedCallback.execute(this);
       }
-      hub.captureTransaction(transaction);
+      hub.captureTransaction(transaction, this.traceState());
+    }
+  }
+
+  @Override
+  public @Nullable TraceState traceState() {
+    if (hub.getOptions().isTraceSampling()) {
+      if (traceState == null) {
+        final AtomicReference<User> userAtomicReference = new AtomicReference<>();
```

Comment:
what's about adding a function to the Scope called `withUser` like we do with `withSession` instead of using `AtomicReference`? that executes a callback and returns the user atomically, it could be package private or `@ApiStatus.Internal`. not a big fun of  `AtomicReference`

Reply:
I agree, also didn't like it, but in this PR we need to avoid introducing changes to the API, so i think `AtomicReference` as a temporary solution is a right trade off.